src/libraries/System.Transactions.Local/src/System/Transactions/DtcProxyShim/DtcInterfaces/IPrepareInfo.cs

```diff
@@ -2,14 +2,15 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
 
 namespace System.Transactions.DtcProxyShim.DtcInterfaces;
 
 // https://docs.microsoft.com/previous-versions/windows/desktop/ms686533(v=vs.85)
-[ComImport, Guid("80c7bfd0-87ee-11ce-8081-0080c758527e"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
-internal interface IPrepareInfo
+[GeneratedComInterface, Guid("80c7bfd0-87ee-11ce-8081-0080c758527e"), InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+internal partial interface IPrepareInfo
 {
     void GetPrepareInfoSize(out uint pcbPrepInfo);
 
-    void GetPrepareInfo([MarshalAs(UnmanagedType.LPArray), Out] byte[] pPrepInfo);
```

Comment:
Was removing this `Out` intentional? If so, why?

Reply:
Yes, `[Out]` does nothing on arrays of blittable values, so we don't allow it.

Reply:
The `[Out]` attribute is incredibly confusing and often causes more trouble than help. @jkoritzinsky or @elinor-fung didn't we right up details on it in the original design doc?

Reply:
I think we did. I’ll make sure on Monday

Reply:
Yep, we have it in our compatibility doc right here: https://github.dev/dotnet/runtime/pull/87351